internal/cluster/service.go

```diff
@@ -355,7 +355,11 @@ func (NotSupportedDistributionError) ServiceError() bool {
 }
 
 func (s service) getDistributionService(cluster Cluster) (Service, error) {
-	service, ok := s.distributions[cluster.Distribution]
+	key := cluster.Distribution
+	if key == "pke" {
+		key += cluster.Cloud
+	}
+	service, ok := s.distributions[key]
```

Comment:
This could go away if the pke cloud distribution was hidden inside the pke service. See my comment above.